Sources/EdgeState.swift

```diff
@@ -74,4 +93,57 @@ class EdgeState {
         currentCollectConsent = status
```

Comment:
This is not related to your PR but some of the variables are not guarded by DispatchQueue and this could cause race conditions and crashes.

Reply:
I've updated EdgeState to guard the consent status. It's better to do it as it keep the design consistent, however I don't think the consent status is accessed from multiple threads. Can you take another look at this part of the PR?